From a6ec4c929ed31bbb2efe383eff53021a09c8983b Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Thu, 27 Oct 2022 10:35:46 +0900
Subject: [PATCH] Solve name conflict with abseil in absl package

When a program in Chromium OS depends on both abseil and libchrome.

Change-Id: I2dbd833347a59fffa8888c817a1fccdc78e5ff0e
patch-name: long-term-0400-absl-thread-annotation-conflict.patch
---
 base/thread_annotations.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/base/thread_annotations.h b/base/thread_annotations.h
index 4fadaeecb6..c6b5533884 100644
--- a/base/thread_annotations.h
+++ b/base/thread_annotations.h
@@ -225,8 +225,11 @@
 // Disables warnings for a single read operation.  This can be used to avoid
 // warnings when it is known that the read is not actually involved in a race,
 // but the compiler cannot confirm that.
-#define TS_UNCHECKED_READ(x) thread_safety_analysis::ts_unchecked_read(x)
+#ifndef TS_UNCHECKED_READ
+#define TS_UNCHECKED_READ(x) libchrome::thread_safety_analysis::ts_unchecked_read(x)
+#endif
 
+namespace libchrome {
 namespace thread_safety_analysis {
 
 // Takes a reference to a guarded data member, and returns an unguarded
@@ -242,6 +245,7 @@ inline T& ts_unchecked_read(T& v) NO_THREAD_SAFETY_ANALYSIS {
 }
 
 }  // namespace thread_safety_analysis
+}  // namespace libchrome
 
 // The above is imported as-is from abseil-cpp. The following Chromium-specific
 // synonyms are added for Chromium concepts (SequenceChecker/ThreadChecker).
-- 
2.42.0.582.g8ccd20d70d-goog

